serverlessworkflow/modules/ROOT/pages/eventing/event-correlation-with-workflows.adoc

```diff
@@ -1 +1,216 @@
+
+[[ref-new-user-account-event]]
+.New Incoming Start Event `newAccountEvent`
+[source,json]
+----
+{
+  "specversion": "0.3",
+  "id": "1d174d25-46ac-4785-bc76-457c2d37d2fe",
+  "source": "",
+  "type": "newAccountEventType",
+  "time": "2022-07-25T16:30:35.461988261-03:00",
+  "userid": "12345",
+  "data": {
+    "email": "test@test.com",
+    "userId": "12345"
+  }
+}
+----
+
+[NOTE]
+====
+It is not supported to correlate multiple events together, events are evaluated with correlations independently.
+====
+
+=== Producing and consuming events with correlation
+
+Following the link:{kogito_sw_examples_url}/serverless-workflow-correlation-quarkus/src/main/resources/correlation.sw.json[workflow definition], it is possible to define other events that are supposed to be published and consumed by this workflow.
+
+The given example uses Callback States, such as the  <<ref-validade-user-email-state,`Validate User Email`>>, indicating once the workflow execution achieves this state, it publishes an event of type <<ref-correlation-event-definition,`validateAccountEmailEvent`>> and wait to receive an event of type <<ref-correlation-event-definition,`validatedAccountEmailEvent`>>. For more information on xref:eventing/working-with-callbacks.adoc[Working with callbacks].
+
+[[ref-validade-user-email-state]]
+.Callback State Definition
+[source,json]
+----
+{
+  "name": "Validate User Email",
+  "type": "callback",
+  "action": {
+    "name": "publish validate event",
+    "eventRef": {
+      "triggerEventRef": "validateAccountEmailEvent"
+    }
+  },
+  "eventRef": "validatedAccountEmailEvent",
+  "transition": "Activate User Account"
+}
+----
+
+[IMPORTANT]
+====
+The produced events contain the same correlation attributes set once the workflow was created.
+====
+
+[[ref-validade-user-email-event]]
+.Produced Callback State event `validateAccountEmailEvent`
+[source,json]
+----
+{
+  "id": "7640a0af-b7fb-4d94-9d9d-3aa1ace60e79",
+  "source": "/process/correlation",
+  "type": "validateAccountEmail",
+  "time": "2022-07-25T16:22:53.735128049-03:00",
+  "data": {
+    "email": "test@test.com",
+    "userId": "12345"
+  },
+  "specversion": "1.0",
+  "kogitoprocinstanceid": "69019826-daef-4fb4-880b-c1658c4e49bc",
+  "kogitoprocid": "correlation",
+  "kogitoprocversion": "1.0",
+  "kogitousertaskist": "1",
+  "kogitoproctype": "SW",
+  "userid": "12345"
+}
+----
+
+All consumed events must contain the same correlation attributes since they are used to identify the workflow instance, in this example
+`userid` with value `12345`, see <<ref-validaded-user-email-event,validatedAccountEmailEvent>>.
+
+[[ref-validaded-user-email-event]]
+.Consumed Callback State event `validatedAccountEmailEvent`
+[source,json]
+----
+{
+  "specversion": "1.0",
+  "id": "953f07a7-aea8-4956-8775-85ab59366fe6",
+  "source": "",
```

Comment:
What about the source? Doesn't it have to be set to `workflow` as is in the workflow definition? Or this is currently not matched at all?

Reply:
@MarianMacik 'source' property is not being considered to match when using correlation, at least on the current impl so in the payload the source could be any value... that's why I kept in this way in the example...

Reply:
OK, thanks.